src/background/services/openPayments.ts

```diff
@@ -904,16 +919,51 @@ export class OpenPaymentsService {
     );
   }
 
+  async manualReconnectWallet() {
+    try {
+      await this.rotateToken();
+    } catch (error) {
+      if (isInvalidClientError(error)) {
+        const msg = this.t('connectWallet_error_invalidClient');
+        throw new Error(msg, { cause: error });
+      }
+      throw error;
+    }
+    await this.storage.setState({ key_revoked: false });
+  }
+
   async reconnectWallet() {
     try {
+      const { walletAddress } = await this.storage.get(['walletAddress']);
+      if (
+        !isWalletAddress(walletAddress) ||
```

Comment:
I think just checking if `walletAddress` exists should be enough.

We can perhaps add such storage validations in `background.ts` based on storage update events - to ensure storage is not manipulated maliciously.